### ISSUE TITLE
Browser Sync vulnerability ... (again)

### ISSUE THREAD

**Issue by snuggs** [devpunks/snuggsi#215]
`axios` vulnerability via `browser-sync`

**Comment by snuggs** [comment]
@brandondees

![tenor](https://user-images.githubusercontent.com/38223/103627398-8e907b00-4f0b-11eb-8172-bd1587590579.gif)

**Comment by brandondees** [comment]
yeah usually dev dependencies are less critical to patch than bits that go into prod, but supply chain attacks and all that are still a factor. I wonder if there's a good pattern for separating the dev kit from the library/package itself, as in separate repos or whatever. That would enable a better signal to noise ratio on tracking changes to snuggsi itself and let the dev tooling be a bit more free to do whatever whenever without bothering downstream users.

**Comment by snuggs** [comment]
@brandondees yeah that's the irony of this project.

99% of any of these headaches don't exist in user land if they use:
```html
<script src=//unpkg.com/snuggsi></script>
```

But people love to install stuff tho. The jQuery folk will get it. That's one thing jQuery got right. Installation process.

**Comment by brandondees** [comment]
@snuggs Yeah I understand. The big downside of linking to a source like unpkg is A: it could have a security breach and the code changes out from under you, B: their downtime is now your downtime too. There's some interesting blockchain / dweb tech coming along that may eventually make those problems a thing of the past, but till then, if we want content integrity, security, and reliability, we have to self-host our frontend.

I'm not arguing that it shouldn't be made available in that form, just that it's totally reasonable/normal for production apps to require their own full copy. The built library doesn't need to ship from NPM with all the dev tools, however. It could be just the one combined source file on npm, and the development setup for it could be installed separately/optionally.

**Comment by snuggs** [comment]
@brandondees makes total sense. Hence this type of rigamarole. I knew from day 1 that unpkg not a final solution. But since didn't actually own `snuggsi.com` at the time had to put is somewhere easily accessible. But figured if good enough for React. Also jQuery still recommends installation from CDN _(while giving instructions for internally)_.

Regarding code change...is this what the `integrity` attribute is for on `<script>`? I knew it's there but never had a use case for it. >>> https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

**Comment by brandondees** [comment]
yep, subresource integrity means the browser will check to make sure the content received matches what the parent page expected before trusting it. best practice, but it also requires that the content from a given asset url never changes unexpectedly (you'd have to have version-specific cdn url to link to before you could safely add the attribute)
